habitat-lab/habitat/datasets/rearrange/samplers/object_sampler.py

```diff
@@ -237,9 +237,13 @@ def sample_placement(
         navmesh_vertices = np.stack(
             sim.pathfinder.build_navmesh_vertices(), axis=0
         )
-        # Note: we cache the largest island to reject samples which are primarily accessible from disconnected navmesh regions. This assumption limits sampling to the largest navigable component of any scene.
-        self.largest_island_size = max(
-            [sim.pathfinder.island_radius(p) for p in navmesh_vertices]
+        # Note: we cache the largest island ID to reject samples which are primarily accessible from disconnected navmesh regions.
+        # This assumption limits sampling to the largest navigable component of any scene.
+        island_size = [
+            sim.pathfinder.island_radius(p) for p in navmesh_vertices
+        ]
```

Comment:
```suggestion
        island_size = list(map(sim.pathfinder.island_radius, navmesh_vertices))
```
Nit is that this might be marginally faster.